### PR TITLE
Add new example for MacOs Install docs

### PR DESCRIPTION
**Pull request by Damans227**
Signed-off-by: Damans227 <Damans227@gmail.com>

### Description of your changes

Adds a new example to install documentation for macOs. Fixes Issue: https://github.com/terraform-docs/terraform-docs/issues/538 

I have:

- [x] Read and followed terraform-docs' [contribution process].
- [x] All tests pass when I run `make test`.

### How has this code been tested

Only test I ran is:  `make test`. It passed completely and generated following comments in the end:

```
coverage: 89.5% of statements
ok  	github.com/terraform-docs/terraform-docs/internal/types	0.203s	coverage: 89.5% of statements
?   	github.com/terraform-docs/terraform-docs/internal/version	[no test files]
?   	github.com/terraform-docs/terraform-docs/scripts/docs	[no test files]
```

[contribution process]: https://git.io/JtEzg

### CODE REVIEW

**Comment by Damans227** [comment]
PTAL @khos2ow

File: docs/user-guide/installation.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 title: "Installation"
 description: "terraform-docs installation guide."
 menu:
-  docs:
-    parent: "user-guide"
+    docs:
```
Comment (khos2ow) [inline docs/user-guide/installation.md:5]: I think these whitespaces might have been added by the editor. Can you revert them back?

File: docs/user-guide/installation.md [hunk 2]
```diff
@@ -117,7 +123,7 @@ autoload -U compinit && compinit
...
 To make this change permanent, the above commands can be added to `~/.profile` file.
 
-[Release]: https://github.com/terraform-docs/terraform-docs/releases
-[Homebrew]: https://brew.sh
-[Scoop]: https://scoop.sh/
-[Chocolatey]: https://www.chocolatey.org
+[release]: https://github.com/terraform-docs/terraform-docs/releases
+[homebrew]: https://brew.sh
+[scoop]: https://scoop.sh/
+[chocolatey]: https://www.chocolatey.org
```
Comment (khos2ow) [inline docs/user-guide/installation.md:129]: Hmm, these should be case sensitive. Was there a reason you changed them?

**Review by khos2ow — COMMENTED** [review]
Thank you @Damans227, couple of small comments below.

File: docs/user-guide/installation.md (hunk 2 above)
Comment (Damans227) [reply docs/user-guide/installation.md:129]: Weird, I didn't even touch those. Pretty sure, my VS Code uses a plugin to format code. May be that caused it. Lemme change it to original formatting.

File: docs/user-guide/installation.md (hunk 1 above)
Comment (Damans227) [reply docs/user-guide/installation.md:5]: yea, it's my editor. I didn't change the whitespaces there. Thanks for noticing them. I will go change it.

**Review by Damans227 — COMMENTED** [review]
I have made the desired changes

**Review by khos2ow — APPROVED** [review]

**Pull request merged**